advanced_samples/Bulk_Upload/DotNet/BulkUploader/Program.cs

```diff
@@ -0,0 +1,219 @@
+                {
+                    try
+                    {
+                        string path = sdsStreamMetaPath + stream.Id + ".json";
+                        Console.WriteLine($"Sending stream metadata from file: {path}");
+                        string meta = File.ReadAllText(path);
+                        metadataService.UpdateStreamMetadataAsync(stream.Id, JsonConvert.DeserializeObject<IDictionary<string,string>>(meta));
+                    }
+                    catch (Exception ex)
+                    {
+                        success = false;
+                        toThrow = ex;
+                    }
+                }
+                if (!String.IsNullOrEmpty(sdsStreamTagPath))
+                {
+                    try
+                    {
+                        string path = sdsStreamTagPath + stream.Id + ".json";
+                        Console.WriteLine($"Sending stream tag from file: {path}");
+                        string meta = File.ReadAllText(path);
+                        metadataService.UpdateStreamTagsAsync(stream.Id, JsonConvert.DeserializeObject<IList<string>>(meta));
+                    }
+                    catch (Exception ex)
+                    {
+                        success = false;
+                        toThrow = ex;
+                    }
+                }
+
+                if (!String.IsNullOrEmpty(sdsStreamDataPath))
+                {
+                    try
+                    {
+                        string path = sdsStreamDataPath + stream.Id + ".json";
+                        Console.WriteLine($"Sending stream data from file: {path}");
+                        string data = File.ReadAllText(path);
+                        var dataAsList = JsonConvert.DeserializeObject<List<JObject>>(data);
+                        dataService.UpdateValuesAsync(stream.Id, dataAsList).Wait();
+                    }
+                    catch (Exception ex)
+                    {
+                        success = false;
+                        toThrow = ex;
+                    }
+                }
+            }
+        }
+
+        private static void SendData()
+        {
+            foreach (String file in Directory.GetFiles(@".", "sdsdata*.json", SearchOption.AllDirectories))
+            {
+                Console.WriteLine($"Sending stream data from file: {file}");
+                string data = File.ReadAllText(file);
+                var matches = Regex.Matches(file, @"(?<=sdsdata)(.+?)(?=.json)");
+                var streamName = matches.First().Value;
+                try
+                {
+                    var dataAsList = JsonConvert.DeserializeObject<List<JObject>>(data);
+                    dataService.UpdateValuesAsync(streamName, dataAsList).Wait();  
+                }
+                catch (Exception ex)
+                {
+                    success = false;
+                    Console.WriteLine(streamName);
```

Comment:
Seems like this was for debugging during development? If desired, likely need to push out to a utility method that's shared by all catch blocks.